solutions/scaling_microfrontend_aws_eks/index.asciidoc

```diff
@@ -0,0 +1,161 @@
+//Category=Kubernetes
+//Product=AWS Elastic Kubernetes Service;
+//Platform=AWS
+//Maturity level=Initial
+
+:toc: macro
+toc::[]
+:idprefix:
+:idseparator: -
+
+= Scaling micro-frontend apps - AWS Elastic Kubernetes Service
+
+== Context & Problem
+=== Microfrontends Introduction
+
+Consider developing a large enterprise application having numerous modules. It makes sense to go with a microservices approach for the backend. It also makes sense to follow a similar approach for the frontend. There are so many elements in the UI that each element can be developed in isolation by a dedicated team. For example, an e-commerce application will have a listing page for the products, a products page to view a single product, a cart element to manage all the products selected by the user, a checkout page to finally purchase the products, etc. Each element has its own identity, that is, it serves an independent purpose of the entire application. You can see how the different elements can be developed independently. You can also partition the application vertically and have dedicated teams for each vertical slice, taking care of its frontend, backend, database and deployment, rather than simply having frontend and backend teams as usual.
+
+There are different approaches to micro-frontend. They usually try to resolve the challenge of how to put together your independently developed elements and present it as a single application to the end user. One approach is to create reusable custom elements by following the https://developer.mozilla.org/en-US/docs/Web/Web_Components[Web Components] standard. In this appraoch you have different teams develop the custom element they are responsible for and put them together in an encapsulating application which is deployed as a single unit, for example using a docker container. You can follow https://github.com/devonfw/devon4ts/wiki/guide-angular-elements[this guide] to developing custom elements in Angular. One challenge with this approach is that when the demand for one of your micro-frontend app increases, you have to increase the number of containers serving your entire application. For example, considering the e-commerce application we mentioned earlier, the products page might get more hits than the checkout page. But even then you have to scale out the containers serving the entire application. This defeats the purpose of developing each element of your application independently.
+
+Another approach, which resolves the above challenge, is to deploy each micro-frontend app in its own container and serve them along different routes of your domain using Ingress. Given the above scenario, when the demand for products page gets higher, you can scale out the containers serving only the products page, rather than scaling out your entire application. 
```

Comment:
I would shorten this heavily.
IMHO we should in the beginning write a short abstract about the solution.
What are you solving how.
This text is full with alternatives and in the end I'm more confused what is coming next.

I would only write 1-2 sentences about what a microfrontend is and link to a website for details (e.g. https://micro-frontends.org/)
Then describes the proposed solution by us

In the end of this article you could create a chapter "Alternatives" where you describe the alternative solutions and their constraints.